ha/fault_tolerance/event_manager.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+
+'''Module whoch provides interfaces to subscribe or unsubscribe for HA
+   events to cortx components'''
+
+import json
+
+from cortx.utils.log import Log
+from ha.core.config.config_manager import ConfigManager
+
+
+class EventManager:
+    '''Module responsible for publishing the HA event to a cortx component
+       who has subscribed for the specific HA event through message bus'''
+
+    def __init__(self):
+        '''Make initialization work for Event Manager'''
+        self._confstore = ConfigManager.get_confstore()
+        self.cortx_comp_list = ['sspl', 'csm', 's3', 'motr', 'hare', 'ha']
+        self._event_key = 'events/subscribe'
+
+    def subscribe(self, component : str, events : list):
+        '''
+           Register all the events for the notification. It maintains
+           list of events registered by the components. It maps the event
+           name with component name and store it in the consul
+           Ex:
+           key: /cortx/ha/v2/events/subscribe/sspl
+           value: ['enclosure:sensor:voltage', ...]
+        '''
+        Log.info(f'Received a subscribe request from {component} for event: {events}')
+        self._validate_component(component)
+        self._validate_events(events)
+        event_string = json.dumps(events)
+        if self._confstore.key_exists(f'self._event_key/{component}'):
+            Log.error(f'{component} has already subscribed')
+            raise Exception(f'{component} has already subescribed')
+        self._confstore.set(f'{self._event_key}/{component}', event_string)
```

Comment:
The component (as key) and the list of events for which it has subscribed (as value) is added here .

We also need to add to confstore the keys based on event names;  for this component

ex  
Key: /cortx/ha/events/<event name> , Value : component name 
This should be done for each event for which this component has registered 


List of components who has subscribed to event 

Key: /cortx/ha/events/<event name> 

Value: [component1, component2, .., componentN] 

Example: 

Key: /cortx/ha/events/enclosure:hw:disk 

Value [csm, hare]

Reply:
Added in PR #513

Reply:
Done